libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -180,6 +180,40 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
         )
     }
 
+    _LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_DEVICE
+    arrival_token arrive_tx(_CUDA_VSTD::ptrdiff_t __arrive_count_update,
+                            _CUDA_VSTD::ptrdiff_t __transaction_count_update) {
+#if (_LIBCUDACXX_DEBUG_LEVEL >= 2)
+        _LIBCUDACXX_DEBUG_ASSERT(__arrive_count_update >= 0);
+        _LIBCUDACXX_DEBUG_ASSERT(__transaction_count_update >= 0);
+        // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#contents-of-the-mbarrier-object
+        // Maximum value of tx-count is 2^20 -1.
+        _LIBCUDACXX_DEBUG_ASSERT(__transaction_count_update <= (1 << 20) - 1);
+        _LIBCUDACXX_DEBUG_ASSERT(__isClusterShared(&__barrier));
+#endif
+        arrival_token __token = {};
+        NV_DISPATCH_TARGET(
+            NV_PROVIDES_SM_90, (
+                if (!__isShared(&__barrier)) {
+                    __trap(); // XXX: We do no support arriving on remote cluster barrier in this method.
+                }
```

Comment:
This should follow the logic structure here: https://github.com/NVIDIA/cccl/blob/830ee33d89e0029a9816a5d993de8dfb9d76e553/libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h#L151-L164C18 

And remove the `  _LIBCUDACXX_DEBUG_ASSERT(__isClusterShared(&__barrier));` above.